plugins/http2_adapter/lib/src/connection_manager_imp.dart

```diff
@@ -4,9 +4,11 @@ part of 'http2_adapter.dart';
 class _ConnectionManager implements ConnectionManager {
   _ConnectionManager({
     Duration? idleTimeout,
+    Duration? handshakeTimeout,
     this.onClientCreate,
     this.proxyConnectedPredicate = defaultProxyConnectedPredicate,
-  }) : _idleTimeout = idleTimeout ?? const Duration(seconds: 1);
+  })  : _idleTimeout = idleTimeout ?? const Duration(seconds: 1),
+        _handshakeTimout = handshakeTimeout ?? const Duration(seconds: 1);
```

Comment:
Is this a proper timeout? How do we evaluate? Do you have any data sources that back this value up?

Reply:
Changed it to be 15 seconds by default. As far as I could tell, this is the average value in all libraries with a small margin; they usually use exactly 10-15 seconds.